rpc/interface/src/rpc_error.rs

```diff
@@ -16,9 +16,16 @@ use serde::{Deserialize, Serialize};
 ///
 /// TODO: This is empty as possible errors will be
 /// enumerated when the handler functions are created.
-#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
-pub enum RpcError {}
+#[derive(Debug, thiserror::Error)]
+pub enum RpcError {
+    /// A [`std::io::Error`] from the database.
+    #[error("database I/O error: {0}")]
+    DatabaseIo(#[from] std::io::Error),
+
+    /// A (non-I/O related) database error.
+    #[error("database error: {0}")]
+    DatabaseError(RuntimeError),
+}
```

Comment:
Change to `cuprate_rpc_interface::RpcError`. The database(s) return `RuntimeError` in their `Service` signatures so we forward that.

This also means that `cuprate_rpc_interface` depends on `cuprate_database`, it doesn't seem ideal to pull all of it in for just the error type though, especially since none of the errors other than IO can happen at this level.

`cuprated` already pulls everything in anyway, but this means other `RpcHandler` impls will have to pull in `cuprate-database` too.

Reply:
Would it worth just making this a `BoxError`? that's what I have done elsewhere, like in the consensus crate.

Reply:
If we're going to be (ab)using opaque errors internally, what about using `anyhow::Error`?

[It's basically a better `Box<dyn Error + Send + Sync + 'static>`](https://www.reddit.com/r/rust/comments/17neomp/comment/k7rhrss/).

Reply:
Yeah sure that's probably a good idea.